packages/core-components/src/lib/unsorted/viz/Dropdown.svelte

```diff
@@ -0,0 +1,60 @@
+<script context="module">
+	export const evidenceInclude = true;
+</script>
+
+<script>
+	import { profile } from '@evidence-dev/component-utilities/profile';
+	import { INPUTS_CONTEXT_KEY } from '@evidence-dev/component-utilities/globalContexts';
+	import { browser } from '$app/environment';
+	import { page } from '$app/stores';
+	import { getContext } from 'svelte';
+
+	const inputs = getContext(INPUTS_CONTEXT_KEY);
+
+	/** @type {string} */
+	export let value;
+
+	/** @type {string} */
+	export let label = value;
+
+	/** @type {string} */
+	export let from;
+	// checks for select and a whitespace after
+	$: if (/select\s/i.test(from)) {
+		from = `(${from})`;
+	}
+
+	/** @type {string} */
+	export let where = 'true';
+
+	/** @type {string} */
+	export let name;
+	$: component_identifier = `_Dropdown-${name}`;
+
+	const search = browser
+		? (query) => profile(db.query, query).then((value) => (data = value))
+		: (query) => (data = profile(db.query, query, component_identifier));
+
+	$: db = $page.data.__db;
```

Comment:
Can we pass functions as part of the page store? 
If so, we should just make the search function part of the page store, that will help keep this from spreading to all components that need it

Reply:
How should the generalized search function look? I think the only possible way to generalize it would be something like `$page.data.search(query, component_identifier).then((value) => data = value)` with a weird synchronous `then` function returned during SSR which imo is less clear

Reply:
Generalized with a third parameter callback so it would look something like `db.query(query, (value) => (data = value))` (for both SSR and CSR) in a new component